Microsoft.Toolkit.Uwp.UI/Extensions/ScrollViewer/ScrollViewerExtensions.MiddleClickScrolling.cs

```diff
@@ -0,0 +1,383 @@
+        }
+
+        private static void ScrollViewer_PointerPressed(object sender, PointerRoutedEventArgs e)
+        {
+            // Unsubscribe if deferred moving is started
+            if (_isDeferredMovingStarted)
+            {
+                UnsubscribeMiddleClickScrolling();
+                return;
+            }
+
+            Pointer pointer = e.Pointer;
+
+            if (pointer.PointerDeviceType == PointerDeviceType.Mouse)
+            {
+                _scrollViewer = sender as ScrollViewer;
+
+                PointerPoint pointerPoint = e.GetCurrentPoint(_scrollViewer);
+
+                // SubscribeMiddle if middle button is pressed
+                if (pointerPoint.Properties.IsMiddleButtonPressed)
+                {
+                    SubscribeMiddleClickScrolling();
+
+                    _startPosition = Window.Current.CoreWindow.PointerPosition;
+                    _currentPosition = Window.Current.CoreWindow.PointerPosition;
+                }
+            }
+        }
+
+        private static void CoreWindow_PointerMoved(CoreWindow sender, PointerEventArgs args)
+        {
+            // If condution that occures before scrolling begins
+            if (_isPressed && !_isMoved)
+            {
+                PointerPoint pointerPoint = args.CurrentPoint;
+
+                if (pointerPoint.Properties.IsMiddleButtonPressed)
+                {
+                    _currentPosition = Window.Current.CoreWindow.PointerPosition;
+
+                    var offsetX = _currentPosition.X - _startPosition.X;
+                    var offsetY = _currentPosition.Y - _startPosition.Y;
+
+                    // Settign _isMoved if pointer goes out of threshold value
+                    if (Math.Abs(offsetX) > _threshold || Math.Abs(offsetY) > _threshold)
+                    {
+                        _isMoved = true;
+                    }
+                }
+            }
+
+            // Update current position of the pointer if scrolling started
+            if (CanScroll())
+            {
+                _currentPosition = Window.Current.CoreWindow.PointerPosition;
+            }
+        }
+
+        private static void CoreWindow_PointerReleased(CoreWindow sender, PointerEventArgs args)
+        {
+            // Start deferred moving if the pointer is pressed and not moved
+            if (_isPressed && !_isMoved)
+            {
+                _isDeferredMovingStarted = true;
+
+                // Event to stop deferred scrolling if pointer exited
+                Window.Current.CoreWindow.PointerExited -= CoreWindow_PointerExited;
+                Window.Current.CoreWindow.PointerExited += CoreWindow_PointerExited;
+
+                // Event to stop deferred scrolling if pointer pressed
+                Window.Current.CoreWindow.PointerPressed -= CoreWindow_PointerPressed;
+                Window.Current.CoreWindow.PointerPressed += CoreWindow_PointerPressed;
```

Comment:
Why subscribe to both CoreWindow and scrollviewer events?

Reply:
`ScrollViewer.PointerPressed` is for `SubscribeMiddleClickScrolling`. `CoreWindow.PointerPressed` is to stop deferred scrolling. I can't use `ScrollViewer.PointerPressed` because I should stop scrolling in `PointerPressed` in the whole app.